### PR TITLE
[formrecognizer] Update generated code for preview.3

### LINKED ISSUE #16682: [formrecognizer] generate code for v2.1-preview.3
Swagger not available yet.

Use multi-api readme: https://github.com/Azure/azure-rest-api-specs/pull/12013/files

Note that there are a few spots in the sync/async _operations_mixin.py generated files that have handwritten code that will need to be preserved until autorest is able to handle multiapi signatures. Example: https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_generated/_operations_mixin.py#L173
The handwritten code is prefixed with FIXME.

### PR DESCRIPTION
**Pull request by catalinaperalta**
Resolves #16682.
Addresses in part #16285.

### CODE REVIEW

**Comment by catalinaperalta** [comment]
Some tests that are still enabled need to be looked into due to preview.3 changes. Also here's the issue tracking uncommenting the tests #16849.

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py [hunk 1]
```diff
@@ -664,7 +711,19 @@ def begin_recognize_custom_forms_from_url(self, model_id, form_url, **kwargs):
...
+        # FIXME: part of this code will be removed once autorest can handle diff mixin
+        # signatures across API versions
+        if pages:
+            if self._api_version == FormRecognizerApiVersion.V2_1_PREVIEW:
+                kwargs.update({"pages": pages})
+            else:
+                raise ValueError(
+                    "'pages' is only available for API version V2_1_PREVIEW and up"
+                )
+
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:726]: qq: will `pages` support for other prebuilts (business card, invoice) be in another PR? I wasn't sure cause I saw the issue tagged.

**Review by kristapratico — APPROVED** [review]
just looks like a few linting errors in CI to resolve but then let's merge! looks great :)

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py [hunk 2]
```diff
@@ -664,7 +711,19 @@ def begin_recognize_custom_forms_from_url(self, model_id, form_url, **kwargs):
...
+        # signatures across API versions
+        if pages:
+            if self._api_version == FormRecognizerApiVersion.V2_1_PREVIEW:
+                kwargs.update({"pages": pages})
+            else:
+                raise ValueError(
+                    "'pages' is only available for API version V2_1_PREVIEW and up"
+                )
+        
         polling_interval = kwargs.pop(
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:727]: actually sorry, just realized, can we make these same changes to the async FormRecognizerClient?

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py (hunk 1 above)
Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:726]: Oh I might have missed them, but at this point I'll open another PR after this one is merged in and address the missing pages support for the other prebuilts. I'll update the description on this PR to not close that issue.

Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:726]: Done, updated the description to not close the issue so I can add the other missing pages support.

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py (hunk 2 above)
Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:727]: Yes will update this in the async client. Thanks for catching that!

Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:719]: Done
